### PR TITLE
Re-implements interactive shell for terraform debug shell

### LINKED ISSUE #171: Fix terrafrom debug shell with merged stacks
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:

When we merged the terraform stacks we did not implement the terraform debug shell again

**What you expected to happen**:

It should get the provider credentials and lauch a shell with the working directory in the terraform code and run terraform init. (It should reuse the terraformWrapper in `/pkg/terraform/terraform.go`):

### PR DESCRIPTION
**Pull request by JoshVanL**
**What this PR does / why we need it**:
Re-implements interactive shell for terraform debug shell

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #171 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
/assign @simonswine

### CODE REVIEW

File: pkg/terraform/terraform.go [hunk 1]
```diff
@@ -272,8 +274,27 @@ func (t *Terraform) Destroy(cluster interfaces.Cluster) error {
...
 func (t *Terraform) Shell(cluster interfaces.Cluster) error {
-	// TODO: needs to be implemented
-	return fmt.Errorf("Shell unimplemented")
+
+	if err := t.terraformWrapper(cluster, "", nil); err != nil {
+		return err
+	}
+
+	cmd := exec.Command(
+		"/bin/sh",
```
Comment (simonswine) [inline pkg/terraform/terraform.go]: As just said this should be come from env['SHELL'] and fallback to /bin/sh

File: pkg/terraform/terraform.go [hunk 2]
```diff
@@ -272,8 +274,32 @@ func (t *Terraform) Destroy(cluster interfaces.Cluster) error {
...
 func (t *Terraform) Shell(cluster interfaces.Cluster) error {
-	// TODO: needs to be implemented
-	return fmt.Errorf("Shell unimplemented")
+
+	if err := t.terraformWrapper(cluster, "", nil); err != nil {
+		return err
+	}
+
+	dir := t.codePath(cluster)
+	envVars, err := t.envVars(cluster)
```
Comment (simonswine) [inline pkg/terraform/terraform.go]: It would be good to get all the environment variables and overwrite only the ones from t.envVars

**Comment by MattiasGees** [comment]
Took over and created a new PR #226

**Pull request closed without merging**